molecule/shell.py

```diff
@@ -38,58 +33,10 @@
 ENV_FILE = '.env.yml'
 
 
-def _get_python_version():  # pragma: no cover
-    return sys.version_info
-
-
-def _get_ansible_version():  # pragma: no cover
-    return ansible.__version__
-
-
-def _supported_python2_version():  # pragma: no cover
-    return _get_python_version()[:2] == (2, 7)
-
-
-def _supported_python3_version():  # pragma: no cover
-    return _get_python_version() >= (3, 6)
-
-
-def _supported_ansible_version():  # pragma: no cover
-    if (distutils.version.LooseVersion(_get_ansible_version()) <=
-            distutils.version.LooseVersion('2.4')):
-        msg = ("Ansible version '{}' not supported.  "
-               'Molecule only supports Ansible versions '
-               "'>= 2.4'.").format(_get_ansible_version())
-        util.sysexit_with_message(msg)
-
-    if _supported_python2_version():
-        pass
-    elif _supported_python3_version():
-        if (distutils.version.LooseVersion(_get_ansible_version()) <
-                distutils.version.LooseVersion('2.4')):
-            msg = ("Ansible version '{}' not supported.  "
-                   'Molecule only supports Ansible versions '
-                   "'>=2.5' with Python version '{}'").format(
-                       _get_ansible_version(), _get_python_version())
-            util.sysexit_with_message(msg)
-    else:
-        msg = ("Python version '{}' not supported.  "
-               'Molecule only supports Python versions '
-               "'2.7' and '>= 3.6'.").format(_get_python_version())
-        util.sysexit_with_message(msg)
-
-
-def _allowed(ctx, param, value):  # pragma: no cover
-    _supported_ansible_version()
-
-    return value
-
-
 @click.group()
 @click.option(
     '--debug/--no-debug',
     default=MOLECULE_DEBUG,
-    callback=_allowed,
```

Comment:
Oh, I didn't realize that it's invoked on debug... If you want to preserve similar behavior you could probably invoke `pip check` here. Given that you *want* it.

Reply:
I'm fine to follow the logic of "let dist metadata solve this".